### PR TITLE
Fallback to four digit random pool name and log INFO when in security sandbox

### PR DESCRIPTION
**Pull request by garricko**
Creating a new PR from scratch because I messed up the rebase of PR #697.

### CODE REVIEW

**Comment by brettwooldridge** [comment]
Merged with minor modifications.  Thanks!

File: src/main/java/com/zaxxer/hikari/HikariConfig.java [hunk 1]
```diff
@@ -1014,13 +1016,29 @@ private void loadProperties(String propertyFileName)
...
-         return next;
+      final String prefix = "HikariPool-";
+      try {
+         // Pool number is global to the VM to avoid overlapping pool numbers in classloader scoped environments
+         synchronized (System.getProperties()) {
+            final String next = String.valueOf(Integer.getInteger("com.zaxxer.hikari.pool_number", 0) + 1);
+            System.setProperty("com.zaxxer.hikari.pool_number", next);
+            return prefix + next;
+         }
+      } catch (AccessControlException e) {
```
Comment (johnou) [inline src/main/java/com/zaxxer/hikari/HikariConfig.java]: Rename variable e to ignore?

File: src/main/java/com/zaxxer/hikari/HikariConfig.java [hunk 2]
```diff
@@ -1014,13 +1016,29 @@ private void loadProperties(String propertyFileName)
...
+         // The SecurityManager didn't allow us to read/write system properties
+         // so just generate a random pool number instead
+         Random random = new Random();
```
Comment (johnou) [inline src/main/java/com/zaxxer/hikari/HikariConfig.java]: ThreadLocalRandom?

Comment (brettwooldridge) [reply src/main/java/com/zaxxer/hikari/HikariConfig.java]: [Already handled](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/HikariConfig.java#L1033) when I merged. :smile:

**Comment by brettwooldridge** [comment]
@garricko HikariCP v2.7.2 containing this fix has been released.  Thanks.

**Pull request merged**